django/core/paginator.py

```diff
@@ -128,6 +131,46 @@ def _check_object_list_is_ordered(self):
 QuerySetPaginator = Paginator   # For backwards-compatibility.
 
 
+class EllipsisPaginator(Paginator):
+    """
+    Paginator with a special method, which builds page range in a compact form
+    - does not display whole range if it does not fit into given size.
+    Instead of full form, it displays beginning, middle and end parts of range
+    with ellipsis between them,
+    e.g.: 1, 2,…, 40, 41, 42, 43, 44, 45, 46, …, 49, 50.
+    Originally used in `django.contrib.admin`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.on_each_side = kwargs.pop('on_each_side', 3)
+        self.on_ends = kwargs.pop('on_ends', 2)
+        self.max_pages_num = kwargs.pop('max_pages_num', 10)
+        super().__init__(*args, **kwargs)
+
+    def get_ellipsed_page_range(self, page_num=1):
+        if self.num_pages <= self.max_pages_num:
+            return self.page_range
+
+        # Insert "smart" pagination links, so that there are always `on_ends`
```

Comment:
I think this comment could do with rewording to actually explain that ellipses are being inserted - the "smart" bit is meaningless and this is not creating the actual links...